apps/test/unit/applab/designModeTest.js

```diff
@@ -165,4 +167,116 @@ describe('setProperty and read Property', () => {
+      // and the background color for that theme:
+      designMode.changeThemeForCurrentScreen(
+        getPrefixedElementById('screen1'),
+        'watermelon'
+      );
+      expect(getPrefixedElementById('screen1')).not.to.be.null;
+      expect(
+        getPrefixedElementById('screen1').getAttribute('data-theme')
+      ).to.equal('watermelon');
+      expect(getPrefixedElementById('screen1').style.backgroundColor).to.equal(
+        'rgb(197, 226, 85)'
+      );
+    });
+
+    it('will change a child of a legacy screen without the data-theme attribute', () => {
+      setExistingHTML(`
+        <div class="screen" id="design_screen1">
+          <input id="design_input1">
+        </div>
+      `);
+
+      // Change theme to default, verify that the screen now has the data-theme attribute
+      // and the textInput now has the padding style and the background color of the new theme:
+      designMode.changeThemeForCurrentScreen(
+        getPrefixedElementById('screen1'),
+        themeOptions[DEFAULT_THEME_INDEX]
+      );
+      expect(getPrefixedElementById('screen1')).not.to.be.null;
+      expect(
+        getPrefixedElementById('screen1').getAttribute('data-theme')
+      ).to.equal(themeOptions[DEFAULT_THEME_INDEX]);
+      expect(getPrefixedElementById('input1')).not.to.be.null;
+      expect(getPrefixedElementById('input1').style.padding).to.equal(
+        '5px 15px'
+      );
+      expect(getPrefixedElementById('input1').style.backgroundColor).to.equal(
+        'rgb(242, 242, 242)'
+      );
+    });
+
+    it('will change a child of a default theme screen', () => {
+      setExistingHTML(`
+        <div class="screen" id="design_screen1" data-theme="default" style="background-color: rgb(255, 255, 255);">
+          <input id="design_input1" style="margin: 0px; width: 200px; height: 30px; border-style: solid; background-color: rgb(242, 242, 242); border-radius: 4px; border-width: 1px; border-color: rgb(77, 87, 95); color: rgb(77, 87, 95); font-family: Arial, Helvetica, sans-serif; font-size: 13px; padding: 5px 15px; position: static; left: 25px; top: 25px;">
+        </div>
+      `);
+
+      // Change theme to default, verify that the screen now has an updated data-theme attribute
+      // and the textInput now has the padding style and the background color of the new theme:
+      designMode.changeThemeForCurrentScreen(
+        getPrefixedElementById('screen1'),
+        'watermelon'
+      );
+      expect(getPrefixedElementById('screen1')).not.to.be.null;
+      expect(
+        getPrefixedElementById('screen1').getAttribute('data-theme')
+      ).to.equal('watermelon');
+      expect(getPrefixedElementById('input1')).not.to.be.null;
+      expect(getPrefixedElementById('input1').style.padding).to.equal(
+        '5px 15px'
+      );
+      expect(getPrefixedElementById('input1').style.backgroundColor).to.equal(
+        'rgb(226, 240, 170)'
+      );
+    });
+  });
```

Comment:
I'd love to see the following test cases:
 - Change a property on an element. Verify that the changed property isn't affected when the theme is changed.
 - Duplicate an element to another screen. Verify that it has the properties of the target screen's theme.
 - ... other cases I'm not thinking of?

Reply:
- Demonstrate that screens/elements with no theme are rendered with the "classic" theme as expected.
- Define expected behavior when the system is presented with unexpected but possible input, such as a screen with a `data-theme` attribute that doesn't match a known theme.
- Test (and confirm that it's intended!) this case where changing themes can clobber a formerly custom style. It'd be good to pin the behavior since it'd be easy to second-guess in the future.
  1. Suppose you have two themes, A and B.  Theme A has a default textinput color 'red' and Theme B has a default textinput color 'blue'.
  2. Create a textinput and set theme A.
  3. Customize the textinput color to 'blue'.
  4. Set theme B. Textinput color is not changed because it wasn't the default.
  5. Set theme A. Textinput color changes back to 'red' because it was the default for B, "losing" the customization.

Some of the special logic I see in the `changeThemeForCurrentScreen` method that could use test coverage:

- Check the special `propShouldUpdate` logic for colors that doesn't apply to other property types.
- For that matter, check that the appropriate properties all have `type: 'color'` defined so they get the special logic.
- Check that `renderDesignWorkspace` isn't called when no properties are changed.
- Maybe something around batching updateProperty calls?

Since theme definitions are still split across many files, they're as much like code as configuration.  I'd like to see tests that verify that all theme definitions are present and complete as expected.  That might include:

- Check that all supported themes that should be shown, do show up in the UI.
- Check that every design element type exports well-formed `themeValues`: Has expected (or at least known-valid) properties listed, and every property has a value for every known theme.
- Clearly more complex, but would give me more confidence: Generate the set of tests that would apply every theme to every element type and verify that every style could be read back from the element afterwards (might catch irrelevant styles defined for an element type).

Reply:
> Check that all supported themes that should be shown, do show up in the UI.

An Eyes test that takes a snapshot after switching between each theme could be low-lift way to implement this last test!